R/user_guide.R

```diff
@@ -40,66 +40,68 @@ create_user_guide <- function(pr) {
 render_user_guide <- function(path, pr) {
   warnings <- warning_message()
 
+  apiSpec <- pr$getApiSpec()
+  title <- apiSpec$info$title
+  version <- apiSpec$info$version
+  title_desc <- "title_desc not generated"
+  body_content <- "body_content not generated"
+
   if (!rlang::is_null(warnings)) {
     warnings <- markdown::markdownToHTML(text = warnings, fragment.only = TRUE)
-    ui <- htmltools::tagList(
-      tags$header(
-        class = "warning",
-        tags$h1(
-          "Warning"
-        ),
-        tags$h2(
-          "The following item(s) need to be resolved before your API will be accessible from Tableau:"
-        )),
-      tags$main(
-        htmltools::HTML(warnings),
-        tags$a(href = "./__docs__/",
-               "Open API documentation")
+
+    title_desc <- htmltools::tagList(
+      tags$h1(
+        class="padded-fully title",
+        title,
+        if (!is.null(version)) paste0("(v", version, ")")
       )
     )
-  } else {
-    apiSpec <- pr$getApiSpec()
 
-    title <- apiSpec$info$title
-    version <- apiSpec$info$version
+    body_content <- htmltools::tagList(
+      tags$h3(
+        class="warning",
+        "Warning: The following item(s) need to be resolved before your API will be accessible from Tableau!"
+      ),
+      tags$div(
+        class="padded-flat-top",
+        htmltools::HTML(warnings)
+      )
+    )
+  } else {
     # Strip description of links to other pages
     desc <- markdown::markdownToHTML(text = strip_md_links(apiSpec$info$description),
                                      fragment.only = TRUE)
-
-    ui <- htmltools::tagList(
-      tags$header(
-        tags$h1(
-          title,
-          if (!is.null(version)) paste0("(v", version, ")")
-        ),
-        tags$div(class = "api-desc",
-                 htmltools::HTML(desc),
-                 tags$p(
-                   tags$a(
-                     href = "./setup",
-                     "Tableau Setup Instructions"
-                   ),
-                   tags$br(),
-                   tags$a(
-                     href = "./__docs__/",
-                     "Open API Documentation"
-                   )
-                 )
-        )
+    
+    title_desc <- htmltools::tagList(
+      tags$h1(
+        class="padded-fully title",
+        title,
+        if (!is.null(version)) paste0("(v", version, ")")
       ),
-      tags$main(
-        tags$div(class = "routes",
-                 lapply(extract_route_info(pr, path), render_route_info)
+      tags$div(class = "api-desc",
+        tags$div(
+          class="padded-flat-top",
+          htmltools::HTML(desc)
         )
       )
     )
+    
+    body_content <- htmltools::tagList(
+      tags$h3(
+        class="subtitle",
+        "Use your analytics extension(s) from Tableau"
```

Comment:
```suggestion
        "Use this extension in Tableau"
```
- I'm torn between "your extension" and "this extension". Leaning towards "this".
- Only need the singular because the generated code is only for this single extension.
- I still don't think this gets across "Auto-generated documentation and usage for _this_ extension. Not sure how we would though. But since this is the first page that shows, I think it'll get the message across.